integrations/messenger/src/utils/oauth.ts

```diff
@@ -0,0 +1,77 @@
+import type { IntegrationContext, Request, Response } from '@botpress/sdk'
+import axios from 'axios'
+import { MessengerClient } from 'messaging-api-messenger'
+import z from 'zod'
+import * as bp from '.botpress'
+
+export class MessengerOauthClient {
+  private clientId: string
+  private clientSecret: string
+
+  constructor() {
+    this.clientId = bp.secrets.APP_ID
+    this.clientSecret = bp.secrets.APP_SECRET
+  }
+
+  async getAccessToken(code: string) {
+    const query = new URLSearchParams({
+      client_id: this.clientId,
+      client_secret: this.clientSecret,
+      redirect_uri: `${process.env.BP_WEBHOOK_URL}/oauth`,
+      code,
+    })
+
+    const res = await axios.get(`https://graph.facebook.com/v18.0/oauth/access_token?${query.toString()}`)
+
+    const data = z
+      .object({
+        access_token: z.string(),
+      })
+      .parse(res.data)
+
+    return data.access_token
+  }
+}
+
+export async function handleOAuthRedirect(req: Request, client: bp.Client, ctx: IntegrationContext): Promise<Response> {
+  const slackOAuthClient = new MessengerOauthClient()
+
+  const query = new URLSearchParams(req.query)
+  const code = query.get('code')
+
+  if (typeof code !== 'string') {
+    throw new Error('Handler received an empty code')
```

Comment:
❓ **question:** 
- The condition does not match the error message, if the condition remains, the message should be "Handle received a non string code".
- If "code" is an empty string, we will not enter the condition, should we check for empty string ?

Reply:
This part was borrowed from the existing oauth logic of Slack, but I agree, the message doesn't make sense for this logic, however, this check isn't needed here either, as the query params are always a string. I changed the logic accordingly here: https://github.com/botpress/botpress/pull/12912/commits/96707605a80c96e7c273c1cd6fd7cbb1c9bda812